hawkbit-core/src/test/java/org/eclipse/hawkbit/api/PropertyBasedArtifactUrlHandlerTest.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.api;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.eclipse.hawkbit.api.ArtifactUrlHandlerProperties.UrlProtocol;
+import org.eclipse.hawkbit.api.URLPlaceholder.SoftwareData;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.common.collect.Lists;
+
+import ru.yandex.qatools.allure.annotations.Description;
+import ru.yandex.qatools.allure.annotations.Features;
+import ru.yandex.qatools.allure.annotations.Stories;
+
+/**
+ * Tests for creating urls to download artifacts.
+ */
+@Features("Unit Tests - Artifact URL Handler")
+@Stories("Test to generate the artifact download URL")
+@RunWith(MockitoJUnitRunner.class)
+public class PropertyBasedArtifactUrlHandlerTest {
+
+    private static final String HTTP_LOCALHOST = "http://localhost:8080/";
+
+    private ArtifactUrlHandler urlHandlerUnderTest;
+
+    private ArtifactUrlHandlerProperties properties;
+
+    private static final Long TENANT_ID = 456789L;
+    private static final String CONTROLLER_ID = "Test";
+    private static final String fileName = "Afile1234";
+    private static final Long softwareModuleId = 87654L;
+    private static final Long targetId = 3474366L;
+    private static final String sha1Hash = "test12345";
+    private static final Long artifactId = 1345678L;
+    private static final String TENANT = "TEST_TENANT";
+
+    private static URLPlaceholder placeholder = new URLPlaceholder(TENANT, TENANT_ID, CONTROLLER_ID, targetId,
+            new SoftwareData(softwareModuleId, fileName, artifactId, sha1Hash));
+
+    @Before
+    public void setup() {
+        properties = new ArtifactUrlHandlerProperties();
+        urlHandlerUnderTest = new PropertyBasedArtifactUrlHandler(properties);
+
+    }
+
+    @Test
+    @Description("Tests the generation of http download url.")
+    public void urlGenerationWithDefaultConfiguration() {
+        properties.getProtocols().put("download-http", new UrlProtocol());
+
+        final List<ArtifactUrl> ddiUrls = urlHandlerUnderTest.getUrls(placeholder, APIType.DDI);
+        assertEquals(
+                Lists.newArrayList(new ArtifactUrl("http", "download-http", HTTP_LOCALHOST + TENANT + "/controller/v1/"
```

Comment:
Extract the list to a constant, so you'll be able to provide a proper name what are you asserting here